processor/k8sprocessor/doc.go

```diff
@@ -16,12 +16,26 @@
 //
 // The processor automatically discovers k8s resources (pods), extracts metadata from them and adds the
 // extracted metadata to the relevant spans, metrics and logs. The processor use the kubernetes API to discover all pods
-// running in a cluster, keeps a record of their IP addresses and interesting metadata. Upon receiving telemetry data,
+// running in a cluster, keeps a record of their IP addresses, pod UIDs and interesting metadata. Upon receiving telemetry data,
```

Comment:
I would restructure the whole section starting from "Upon receiving telemetry data,..."

Like:

The rules for associating the source record with specific Pod Metadata are configured via `pod_association` key. It represents a list of rules that are executed in order until the first one is able to do the match.  Each rule is specified as a pair of `from` (representing the rule type) and `name` (representing the extracted key name). Following rule types are available:

* `from: labels` - takes the attribute from resource attributes (the value can contain either IP address or Pod UID)
* `from: connection` - takes the IP attribute from connection context (if available) and automatically associates it with `k8s.pod.ip` attribute